### PR TITLE
Added Product pages 

### PR DESCRIPTION
**Pull request by rahulcs754**
**Summary**

- Added Product Page
- Added Cart Context and cart Reducer
- added Wish list and wishlist Reducer 
- added filter Context and filter reducer
- Created Some utility function 
    1. sort by category 
    2. sort by price
    3. sort by rating
    4. compose function

- Header change  
   Separated some header part 

image

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ export const products = [
     title: "FERARI PILOTA",
     price: "600",
     mrpPrice: "1200",
-    discount: "50%",
+    discount: "50% off",
     rating: 3,
```
Comment (raman1999) [inline src/backend/db/products.js:89]: you can calculate discount directly from mrp and price, so no need of this

**Review by raman1999 — APPROVED** [review]
code looks clean to me,   good use of reducers

File: src/backend/db/products.js (hunk 1 above)
Comment (rahulcs754) [reply src/backend/db/products.js:89]: right it will change in next commit

**Review by SimranjitKaurMaan — APPROVED** [review]
Good abstraction of components and use of context and reducers. Good to merge

**Pull request merged**